server/handlers/project_command_output_handler.go

```diff
@@ -114,6 +116,15 @@ func NewAsyncProjectCommandOutputHandler(
 	}
 }
 
+func (p *AsyncProjectCommandOutputHandler) IsKeyExists(key string) bool {
+	p.receiverBuffersLock.RLock()
+	defer func() {
+		p.receiverBuffersLock.RUnlock()
+	}()
```

Comment:
don't think you need to wrap it into anonymous function here.

Reply:
Yup! I've removed it :)